doc/ExternalDependencies.md

```diff
@@ -0,0 +1,60 @@
+External Dependencies
+=====================
+
+<!-- npm i -g markdown-toc; markdown-toc -i ExternalDependencies.md -->
+
+<!-- toc -->
+
+- [Adding a dependency](#adding-a-dependency)
+  * [Provide Important Functionality](#provide-important-functionality)
+  * [Project maturity](#project-maturity)
+  * [Limited dependencies](#limited-dependencies)
+- [Dependency Tree](#dependency-tree)
+
+<!-- tocstop -->
+
+## Adding a dependency
+
+External dependencies can be added to Tock on a case by case basis. Each
+dependency will be reviewed for inclusion. Below is a general list of
+requirements for adding a new external crate.
+
+### Provide Important Functionality
+
+The external crate must provide important functionality that couldn't
+easily or realisticly be provided by the Tock developers.
+
+Examples of this include:
+
+ * Crypto libraries
+     * No one should write their own crypto library
+ * Wireless protocols
+     * Wireless implementations are difficult to get the correct timing
+     * Wireless protocols are also very expensive to certify
+ * Rust embedded HALs
+     * For some platforms there are mature Rust Embedded HALs. This would take
+       a long time for Tock developers to re-implement themselves.
+
+### Project maturity
+
+The external crate being added must be a mature project, with a high quality
+of code. The project must be well regarded in the Rust community.
```

Comment:
What do we typically look at to see if a crate is `high quality`?

Reply:
It's kind of vague I know, but I don't really have a good way to describe whats "high quality"

Reply:
I think it'll just be up to reviewers